blocks/bricks/__init__.py

```diff
@@ -158,7 +159,56 @@ def _push_initialization_config(self):
                     child.biases_init = self.biases_init
 
 
-class Linear(Initializable):
+class Feedforward(Brick):
+    """Declares an interface for bricks with one input and one output.
+
+    Many bricks have just one input and just one output (activations,
+    :class:`Linear`, :class:`MLP`). To make such bricks interchangable
+    in most contexts they should share an interface for configuring
+    their input and output dimensions. This brick provides such an
+    interface.
+
+    """
+    @abstractproperty
+    def input_dim():
+        """The input dimension of the brick."""
+        pass
+
+    @abstractproperty
+    def output_dim():
+        """The output dimension of the brick."""
+        pass
+
+
+def override_with_attribute(property_, attribute=None):
```

Comment:
Because `input_dim` is deliberately made an abstract property so that every attempt to access it would cause a failure. I did not include a default interpretation in the `Feedforward` to prevent mistakes when people forget to override the required properties. `override_with_attribute` overrides an abstract property with a very simple ones, behaving exactly like an attribute.